src/averageAges.js

```diff
@@ -55,10 +80,22 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // 1. find a mother of each person (or only for men)
-  // 2. keep people who have mothers in the array
-  // 3. calculate the difference child.born - mother.born
-  // 4. return the average value
+  const children = people.filter(child => {
+    return onlyWithSon
+      ? people.find(person => child.sex === SEX_MALE
+        && child.mother === person.name)
+      : people.some((person) => {
+        return child.mother === person.name;
+      });
+  });
```

Comment:
Now everything is written correctly, but it is difficult to read. 
Try first to filter separately the array of `people` and leave only those objects that have a `mother`. 
And then check if `onlyWithSon` - true, filter the objects that have a mother and leave only those that are men.